enterprise/views.py

```diff
@@ -269,12 +270,31 @@ def get(self, request):
             raise Http404
 
         item = 'course' if course_id else 'program'
+
+        enterprise_customer = consent_record.enterprise_customer
+        if course_id:
+            try:
+                catalog_api_client = CourseCatalogApiServiceClient(enterprise_customer.site)
+            except ImproperlyConfigured:
+                raise Http404
+
+            course_run_details = catalog_api_client.get_course_run(course_id)
+            course_start = parse(course_run_details['start'])
+
         enterprise_customer = consent_record.enterprise_customer
         context_data = self.get_default_context(enterprise_customer, request)
         context_data.update(
             {
                 'course_id': course_id,
                 'course_specific': True,
+                'course_title': course_run_details['title'],
+                'course_start_date': _(
+                    '{month} {day}, {year}'
+                ).format(
+                    month=month_name[course_start.month],
+                    day=course_start.day,
+                    year=course_start.year,
```

Comment:
This will cause the page to choke if we are in the context of a program because `course_start` will not be defined.

Reply:
Actually, I would just refactor this all into the `if course_id` block you have above on line 275.